Window (default)

ethmth opened the pull request:
Solved the following issues:

> This window contains the site's hero message.
Window is draggable only in desktop view
Windows is open when site loads
Window can be closed only in desktop view
"DevFest 23" Icon opens window

In order to implement the functionality of having the Icon make a window open/close, I made the Window component a child of Icon component.

I made both components abstract enough using props that one should not have to edit them directly in order to create a new Window/Icon pair. The DefaultWindow component is a good example of how to use the Icon component to create a Window under my implementation. I thought this would make it easier for the people who have to create more windows in the future, but it can be kinda limiting based on the props I implemented, so I can revert to having a unique component for each window/icon if needed.

I also added Mobile detection and changed up the functionality/styling depending on whether a mobile or desktop device is detected. This seems to work quite well on Desktop/Phone, but the react library I used for the Mobile detection itself detects an iPad as Mobile not Desktop, and it doesn't look so great. I can fix this in the future but probably not by tomorrow because I'm busy. 

Yea I don't really know what else to put here so let me know if I need to change anything or fix it. I'd be happy to do so.

ModeAxe commented:
Bug:

Z-space issue when multiple windows open
![image](https://user-images.githubusercontent.com/41904782/211701273-df5ecf3f-cce5-489c-b3db-1c33d237a708.png)

ethmth commented:
Just pushed some fixes:

- Fixed the Z-index issue for multiple windows
- Fixed unique styling issues for multiple windows by utilizing styled components
- Changed naming convention (DefaultWindow -> WindowDefault) so that Windows will show up next to each other lexicographically. 
- Created WindowTest as a second window for multiple window testing.

Remaining issues:
- iPad is still detected as a mobile device, not desktop, and the formatting on iPad doesn't look great since mobile formatting is targeted towards a vertical device.
- The desktop height is capped to the view height. This is great for desktop because you can't drag things off screen, but on mobile it means you can't scroll down to view further windows.

ModeAxe commented on src/Components/Desktop.jsx:
```diff
@@ -1,13 +1,18 @@
+import { isMobile } from "react-device-detect";
```
Have you considered using viewport size/aspect ratio to determine is "mobile" or not? This should fix the problem with iPad too since the biggest issue is rendering a vertical version (windows open, scrolling, no drag, no icons etc) of the site and a horizontal version (desktop experience). Perhaps calling it "mobile" and not "vertical" in the design doc was a mistake since it's more about the orientation of the page than the device used. 
Using viewport will also mean if someone rotates their phone, they view the desktop experience.

ModeAxe commented:
> * The desktop height is capped to the view height. This is great for desktop because you can't drag things off screen, but on mobile it means you can't scroll down to view further windows.

Capped desktop height is because of the `overflow: hidden` in the desktop's CSS. Changing that to 'overflow: auto' when vertical allows you to scroll.

ethmth commented:
Thanks for the help. I think I was able to implement proper functionality in the latest commit. Let me know if there's anything else I need to fix!

ModeAxe commented:
Looks good! Feel free to merge it into the `react-site` branch. 
There are a couple of styling issues but we can address those in a final pass once functionality is done.
